source/parser/independent/intermediate-macro-instruction.cpp

```diff
@@ -98,8 +99,36 @@ void IntermediateMacroInstruction::execute(
     CompileErrorList& errors,
     FinalCommandVector& commandOutput,
     MemoryAccess& memoryAccess) {
+  CompileErrorList innerErrors;
+  FinalCommandVector innerCommands;
   for (const auto& operation : _operations) {
-    operation->execute(immutable, errors, commandOutput, memoryAccess);
+    operation->execute(immutable, innerErrors, innerCommands, memoryAccess);
+  }
+  // If the inner commands don't have their own position, use the position of
```

Comment:
Do we really only want to do this for the build-in macros? (as far as I could read that out of your code)
What, if the user designs a small macro, but then uses it with wrong syntax like 100 times? Wouldn't that cause a huge bunch of error messages in the macro definition? I'm rather a fan of showing it at the macro call position consequently of each call. Any objections to this?
I mean, the best would be to see the errors in the macro expansion window (and at the instruction itself), but it is probably too little time to do that right now..